libmscore/score.cpp

```diff
@@ -2067,59 +2034,132 @@ bool Score::appendScore(Score* score, bool addPageBreak, bool addSectionBreak)
+            Staff* staff = this->staff(staffIdx);
+            Staff* ostaff = score->staff(staffIdx);
+            
+            // check if key signature needs to be changed
+            if (!(ostaff->key(otick) == staff->key(ctick))) {
+
+                  Segment* ns = firstAppendedMeasure->undoGetSegment(SegmentType::KeySig, ctick);
+                  KeySigEvent nkse = KeySigEvent(ostaff->keySigEvent(otick));
+                  KeySig* nks = new KeySig(this);
+                  nks->setScore(this);
+                  nks->setTrack(trackIdx);
+
+                  nks->setKeySigEvent(nkse);
+                  staff->setKey(ctick, nkse);
+                  ns->add(nks);
+                  //undoAddElement(nks);
+                  }
+            // check if a key signature is present but is spurious (i.e. no actual change)
+            else if (staff->currentKeyTick(ctick) == ctick &&
+                     staff->key(ctick - Fraction::fromTicks(1)) == ostaff->key(otick)) {
+                        Segment* ns = firstAppendedMeasure->first(SegmentType::KeySig);
+                        if (ns) ns->remove(ns->element(trackIdx));
+                  }
+            
+            // check if time signature needs to be changed
+            if (!ostaff->timeSig(otick) || !staff->timeSig(ctick)) {} // yes, this can happen.
+            else if (!((*ostaff->timeSig(otick)) == (*staff->timeSig(ctick)))) {
+                  Segment* ns = firstAppendedMeasure->undoGetSegment(SegmentType::TimeSig, ctick);
+                  TimeSig* nsig = new TimeSig(*(ostaff->timeSig(otick)));
+
+                  nsig->setScore(this);
+                  nsig->setTrack(trackIdx);
+                  ns->add(nsig);
+                  //undoAddElement(nsig);
                   }
+            // check if a time signature is present but is spurious (i.e. no actual change)
+            else if (staff->currentTimeSigTick(ctick) == ctick &&
+                     (*staff->timeSig(ctick - Fraction::fromTicks(1))) == (*ostaff->timeSig(otick))) {
+                        Segment* ns = firstAppendedMeasure->first(SegmentType::TimeSig);
+                        if (ns) ns->remove(ns->element(trackIdx));
+                  }
+
+            // check if clef signature needs to be changed
+            if (!(ostaff->clef(otick) == staff->clef(ctick))) {
+                  undoChangeClef(staff,firstAppendedMeasure,ostaff->clef(otick));
+                  }
+            // check if a clef change is present but is spurious (i.e. no actual change)
+            else if (staff->currentClefTick(ctick) == ctick &&
+                     staff->clef(ctick - Fraction::fromTicks(1)) == ostaff->clef(otick)) {
+                        Segment* ns = firstAppendedMeasure->first(SegmentType::Clef);
+                        if (!ns) ns = firstAppendedMeasure->first(SegmentType::HeaderClef);
+                        if (ns) ns->remove(ns->element(trackIdx));
```

Comment:
new line for bodies. For the record, I'm only going to mention this for bodies that have some complexity. I don't really mind if you put `break` or `continue` on the same line as a simple if statement.